src/core/ipfs/files.js

```diff
@@ -21,13 +23,18 @@ module.exports = function files (self) {
       const i = new Importer(self._dagS)
       const res = []
 
-      i.on('data', (info) => {
-        res.push(info)
-      })
-
-      i.once('end', () => {
+      // Transform file info tuples to DAGNodes
+      i.pipe(through.obj(function transform (info, enc, next) {
+        console.log('info', info)
+        self._dagS.get(bs58.encode(info.multihash).toString(), (err, node) => {
```

Comment:
Yeah, I ended up refactoring everything to include the path, so that instead of returning DAGNodes everywhere we give back `{ path: '...', node: DAGNode }`.